client/src/components/OpportunitiesList.js

```diff
@@ -90,24 +89,81 @@ class OpportunitiesList extends Component {
     });
   };
   handleSelectCity = (e, data) => {
-    const selectedCity = data.value;
+    const City = data.value;
     this.setState({
-      formEntries: { ...this.state.formEntries, city: selectedCity }
+      selectedCity: City
     });
   };
-  filteringOpportunitiesBySkills = () => {
-    const filteredBySkills = this.state.OpportunitiesList.filter(opportunity =>
-      this.state.selectedSkills.some(skill =>
-        opportunity.skills.includes(skill)
-      )
-    ).filter(opportunity => this.state.selectedJobType === opportunity.type);
+  handleChangeSearchKeyWord = e => {
+    const searchKeyWord = e.target.value;
     this.setState({
-      filteredResult: filteredBySkills,
-      filtering: true
+      ...this.state,
+      searchKeyWord: searchKeyWord
     });
   };
+  filterOpportunities = () => {
+    const {
+      selectedCity,
+      searchKeyWord,
+      selectedJobType,
+      opportunitiesList,
+      selectedSkills
+    } = this.state;
+    if (!selectedCity && !selectedJobType && searchKeyWord) {
+      return opportunitiesList.filter(opportunity => {
+        return (
+          (opportunity.description &&
+            opportunity.description
+              .toLowerCase()
+              .includes(searchKeyWord.toLowerCase())) ||
+          (opportunity.opportunity_title &&
+            opportunity.opportunity_title
+              .toLowerCase()
+              .includes(searchKeyWord.toLowerCase()))
+        );
+      });
+    }
+    if (selectedCity && !selectedJobType) {
+      return opportunitiesList.filter(opportunity => {
+        return selectedCity.includes(opportunity.location);
+      });
+    }
+    if (!selectedCity && selectedJobType) {
+      return opportunitiesList.filter(opportunity => {

```

Comment:
if the arrow function has only one statement, then you can drop the `{` `}` and the `return`

Reply:
same for other ones in this file